src/modules/blockchainApplication/manage/components/RemoveApplicationDetails/RemoveApplicationDetails.js

```diff
@@ -61,85 +94,131 @@ const RemoveApplicationDetails = ({ location, application, onCancel, nextStep })
   ];
 
   const handleRemoveApplication = () => {
-    // TODO: this needs to be reinstated when the set application flow is completed
-    // because presently, there is no way to set current application on runtime
-
-    // deleteApplicationByChainId(chainID);
+    deleteApplicationByChainId(chainId);
     nextStep({ application });
   };
 
+  if (isOnChainDataError || isOffChainDataError)
+    return (
+      <Dialog hasClose className={`${grid.row} ${grid['center-xs']}`}>
+        <div className={`${styles.wrapper} ${styles.errorWrapper}`}>
+          <Illustration name="applicationDetailsError" />
+          <div className={styles.errorText}>{t("Couldn't load application data")}</div>
```

Comment:
"Couldn't load application data" sounds too informal. Could you replace it with "Error loading application data"?

Reply:
@reyraa Fixed